core/object/object.cpp

```diff
@@ -519,8 +519,14 @@ void Object::get_property_list(List<PropertyInfo> *p_list, bool p_reversed) cons
 	for (const KeyValue<StringName, Variant> &K : metadata) {
 		PropertyInfo pi = PropertyInfo(K.value.get_type(), "metadata/" + K.key.operator String());
 		if (K.value.get_type() == Variant::OBJECT) {
+			Object *obj = K.value;
 			pi.hint = PROPERTY_HINT_RESOURCE_TYPE;
-			pi.hint_string = "Resource";
+			if (obj->is_class("Script")) {
```

Comment:
Is it technically possible for `obj` to be `nullptr`?

Reply:
I'd probably add a condition like 
```suggestion
			if (obj && obj->is_class("Script")) {
```

Reply:
That code will use `Variant::operator Object *` that doesn't validate the `Object` so no nulls would come from a `Variant` of type object, in principle. However, I think it's a valid state for `Variant` to the of type object but pointing to null, so I'm not really sure...

That said, `if (Object::cast_to<Script>(obj))` seems to be a better fit here anyway.

Reply:
@RandomShaper Fixed!

If one of you could approve the PR, it could be nice!